### PR TITLE
nimbus author inherent mock

### PR DESCRIPTION
**Pull request by ermalkaleci**

### CODE REVIEW

File: packages/core/src/utils/time-travel.ts [hunk 1]
```diff
@@ -12,7 +12,8 @@ export const getCurrentSlot = async (chain: Blockchain) => {
   // use raw key here because some chain did not expose those storage to metadata
   const slotRaw = meta.consts.babe
     ? await chain.head.get('0x1cb6f36e027abb2091cfb5110ab5087f06155b3cd9a8c9e5e9a23fd5dc13a5ed') // babe.currentSlot
-    : await chain.head.get('0x57f8dc2f5ab09467896f47300f04243806155b3cd9a8c9e5e9a23fd5dc13a5ed') // aura.currentSlot
+    : (await chain.head.get('0x57f8dc2f5ab09467896f47300f04243806155b3cd9a8c9e5e9a23fd5dc13a5ed')) || // aura.currentSlot
```
Comment (xlc) [inline packages/core/src/utils/time-travel.ts:15]: make this into a helper method and make it a bit more readable

**Comment by ermalkaleci** [comment]
@girazoki Does your system  require `authoritiesNoting.setLatestAuthoritiesData` to be called on every block? This change will allow your chain to build blocks but I don't know if all functionalities will work. Can you help with testing?

**Comment by ermalkaleci** [comment]
@girazoki test latest beta `0.9.8-2` and let me know if it works

**Comment by girazoki** [comment]
@ermalkaleci we have not been able to use directly the version you point at:

`npx @acala-network/chopsticks@0.9.8-2 --help` throws:

onst _jsondiffpatch = require("jsondiffpatch");
                       ^

```
Error [ERR_REQUIRE_ESM]: require() of ES Module /home/girazoki/Desktop/tanssi/node_modules/.pnpm/jsondiffpatch@0.6.0/node_modules/jsondiffpatch/lib/index.js from /home/girazoki/Desktop/tanssi/node_modules/.pnpm/@acala-network+chopsticks@0.9.8-2_debug@4.3.4/node_modules/@acala-network/chopsticks/dist/cjs/utils/decoder.js not supported.
Instead change the require of index.js in /home/girazoki/Desktop/tanssi/node_modules/.pnpm/@acala-network+chopsticks@0.9.8-2_debug@4.3.4/node_modules/@acala-network/chopsticks/dist/cjs/utils/decoder.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/girazoki/Desktop/tanssi/node_modules/.pnpm/@acala-network+chopsticks@0.9.8-2_debug@4.3.4/node_modules/@acala-network/chopsticks/dist/cjs/utils/decoder.js:12:24) {
  code: 'ERR_REQUIRE_ESM'
}
```
However after hacking a bit in the version branch we have been able to make it work and we confirm that is able to fork correctly our networks. We might need to propose one additional inherent to make it fully compatible but we can do this, as right now it seems much easier.

Thanks!

**Comment by ermalkaleci** [comment]
@girazoki I've pushed a new beta with the fix

**Comment by girazoki** [comment]
I confirm it works with our container-chains, thank you so much! As I said for the tanssi chain we need an additional inherent, but we should be able to do a PR for it. Will let you know when we have it!

**Comment by ermalkaleci** [comment]
Yes, with the refactored inherent it should be straightforward

**Comment by girazoki** [comment]
https://github.com/AcalaNetwork/chopsticks/pull/665 CC: @ermalkaleci

**Pull request merged**